Support multiple services as specified in the Presentation API v2.1

abollini opened the pull request.

abollini committed:
fix initialization with empty constructor

abollini committed:
allow multi services as in version 2.1

abollini committed:
add support for multiple services to all the resource types

abollini committed:
Revert "add support for multiple services to all the resource types"

This reverts commit d4f9dc55bb52d615a05658eb44bb95d7fb18a830.

abollini committed:
add support for nested service in service as required for autocomplete on iiif search api 1.0 service

abollini commented:
I have revert the commit d4f9dc5 as adding support for multiple services everywhere change the serialization of the manifest and this break some tests other than most probably many client. The IIIF examples show just a single service for
- images
- thumbnails
- service (IIIF Search API, a search service can have a nested service for autocomplete)

with these changes I'm able to generate manifest like this one http://uvdemo.4science.it/manifest/hamlet

jbaiter commented on iiif-presentation-model-api/src/main/java/de/digitalcollections/iiif/presentation/model/api/v2/IiifResource.java:
```diff
@@ -64,7 +64,7 @@
   void setSeeAlso(List<SeeAlso> seeAlso);
 
-  Service getService();
+  List<Service> getService();
```
Minor nitpick, but this should probably be `getServices` now that multiple services are supported

abollini committed:
update getter/setter to reflect the support for multiple services

abollini committed:
Merge branch 'master' of https://github.com/dbmdz/iiif-presentation-api into support21

abollini committed:
fix test failure and javadoc after merge with the updated master

abollini commented:
Hi,
sorry I was unable to immediately apply the suggested changes and I have forget about that for a while...
I have renamed the methods to reflect the plural of services and updated the relevant test so that the build success
